src/api.ts

```diff
@@ -25,8 +25,7 @@ router.post('/*', async (req, res) => {
   if (!url) return subgraphError(res, 'No subgraph URL provided', 400);
   if (!query) return subgraphError(res, 'No query provided', 400);
 
-  url = url.startsWith('http') ? url : `https://${url}`;
-
+  url = buildURL(url);
```

Comment:
```suggestion
  try {
    url = buildURL(url);
  } catch (error: any) {
    return subgraphError(res, error.message, 400);
  }
```

Reply:
But i think we still need to support other URLs than delegation URLs. maybe we get more subgraphs in the future?

If we want to remove support for that, we have to do it after merging this PR and changing it here https://github.com/snapshot-labs/snapshot.js/blob/master/src/delegationSubgraphs.json else we end up with failing requests